src/twisted/protocols/test/test_tls.py

```diff
@@ -1819,3 +1833,77 @@ def acceptableProtocols(self):
         @rtype: L{list} of L{bytes}
         """
         return self._acceptableProtocols
+
+
+class _AggregateSmallWritesTests(SynchronousTestCase):
+    """Tests for ``_AggregateSmallWrites``."""
+
+    @given(
+        st.lists(
+            st.one_of(
+                st.none(),
+                st.integers(min_value=1, max_value=100_000).map(
```

Comment:
This is more of a philosophical comment in the context of Twisted dev process.

We don't have much time dedicated to doing code reviews.

I had to read this code a few times to understand what is going on here and how the test is structured.

I think that this can discourage a review.

Someone might not have much to do a review, will read the code and will just stop doing the review.

I think that a lot of the contribution to Twisted are done as a 2nd job, without too much time dedicated to doing code reviews.

Now you can argue that if you don't have time to do top quality code review, is better not do do it ... but then we end up with nobody doing reviews.


-------

I understand that the current test input generate is comprehensive and define in a brief way.
I find it hard to read.

I think is better to have comment close this definition.

For example, I don't understand why we need this whole integer to bytes mapping here.
Why not use `st.binary()` ?

Reply:
I didn't do `st.binary()` because my first though is it means it spends time searching the input space random bytes, and this test isn't about the contents of the bytes, it's about the lengths.